File: typescript-connector/src/models/Weight.ts
// Class from DFC Business Ontology: #Weight
import { SemanticObject } from "../core/SemanticObject.js";
import { QuantitativeValue, type QuantitativeValueParams } from "./QuantitativeValue.js";

export interface WeightParams extends QuantitativeValueParams {}
export class Weight extends QuantitativeValue {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Weight";
  }



  constructor(
    semanticId: string,
    params?: WeightParams,
  ) {
    super(semanticId, params);
    this.semanticType = Weight.SEMANTIC_TYPE;
  }
  static {
    SemanticObject.typeRegistry.set(Weight.SEMANTIC_TYPE, Weight);
  }
}

File: ruby-gem/lib/models/weight.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Weight
require_relative '../core/semantic_object'
require_relative 'quantitative_value'


module DfcLinkmlConnector
  module Models
    class Weight < QuantitativeValue
      SEMANTIC_TYPE = "dfc-b:Weight".freeze

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, value: nil, unit: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, value: nil, unit: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension, value: value, unit: unit)

        self.semanticType = "dfc-b:Weight"

      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
